src/middleware.ts

```diff
@@ -264,6 +264,19 @@ export function* simpleCache<Ctx extends ApiCtx = ApiCtx>(
   ctx.actions.push(addData({ [key]: data }));
 }
 
+export function* customKey<Ctx extends ApiCtx = ApiCtx>(ctx: Ctx, next: Next) {
+  if (
+    ctx?.key &&
+    ctx?.action?.payload?.key &&
+    ctx.key !== ctx.action.payload.key
+  ) {
+    const newKey = ctx.name.split('|')[0] + '|' + ctx.key;
+    ctx.key = newKey;
+    ctx.action.payload.key = newKey;
+  }
+  yield next();
+}
+
```

Comment:
Is there any reason why we wouldn't add `customKey` to the `requestMonitor` middleware by default?  This seems like a pretty safe addition and makes it easier for end users to use it.

Reply:
No reason. However, I wanted this decision to be yours.